pkg/cli/api_server.go

```diff
@@ -67,16 +68,20 @@ func (c *APIServerCommand) Flags() *cli.FlagSet {
 
 func (c *APIServerCommand) Run(ctx context.Context, args []string) error {
 	server, grpcServer, closer, err := c.RunUnstarted(ctx, args)
+	defer func() {
+		if err := closer.Close(); err != nil {
+			logging.FromContext(ctx).Errorw("failed to close", "error", err)
+		}
+	}()
 	if err != nil {
 		return err
 	}
-	defer closer()
 
 	return server.StartGRPC(ctx, grpcServer)
 }
 
-func (c *APIServerCommand) RunUnstarted(ctx context.Context, args []string) (*serving.Server, *grpc.Server, func(), error) {
-	closer := func() {}
+func (c *APIServerCommand) RunUnstarted(ctx context.Context, args []string) (*serving.Server, *grpc.Server, *multicloser.Closer, error) {
+	closer := &multicloser.Closer{}
```

Comment:
Nit: 

```suggestion
	var closer *multicloser.Closer
```

(throughout)

Reply:
Wouldn't it be safer to do `closer := &multicloser.Closer{}`? Otherwise returning a `nil` closer and calling `closer.Close()` will panic. We don't have safe func to call to handle nil

Reply:
Maybe we do have safe func... https://github.com/abcxyz/pkg/blob/main/multicloser/multicloser.go#L73

Reply:
I engineered it to be safe 😄

Reply:
Done